### DISCUSSION TITLE
Two recurring errors

### DISCUSSION THREAD

**Discussion by dhmiller** [deforum-art/deforum-stable-diffusion#162] [Q&A]
I continue to get one of two different errors while running Automatic/3D/Hybrid and am not sure if they are Automatic or Deforum based issues (how to tell?) Both stop the render well short of the final keyframe/maximum frames but at different points if I simply rerun the job. I am using multiple keyframes, but have simplified those down to a single kf, yet the problems persist. Any help would be appreciated. 
1 (Deforum extension for auto1111 — version 2.2b | Git commit: ab8650a3 (Sat Mar 11 10:52:16 2023):   UnboundLocalError: local variable 'value_is_number' referenced before assignment

2.RecursionError: maximum recursion depth exceeded while calling a Python object

The most recent crash just now crashed just 9 minutes into the job. That's the shortest one yet. 
Not sure if this helps:    File "C:\Users\dhaym\AppData\Roaming\Visions of Chaos\Examples\MachineLearning\Text To Image\stable-diffusion-webui\venv\lib\site-packages\torch\cuda\__init__.py", line 160, in is_initialized
    return _initialized and not _is_in_bad_fork()
RecursionError: maximum recursion depth exceeded while calling a Python object

RecursionError: maximum recursion depth exceeded while calling a Python object
Time taken: 9m 22.17sTorch active/reserved: 8069/9690 MiB, Sys VRAM: 12323/49140 MiB (25.08%)

Interesting thing is, this is intermittent - if I actually finish a job, I can most likley complete another one if I leave the UI open. But then after some time, it will crash again with one of the errors, using the "maximum recursion" one.

**Comment by dhmiller** [comment]
Anyone have any thoughts on these errors?
I continue to get one of two different errors while running Automatic/3D/Hybrid and am not sure if they are Automatic or Deforum based issues (how to tell?) Both stop the render well short of the final keyframe/maximum frames but at different points if I simply rerun the job. I am using multiple keyframes, but have simplified those down to a single kf, yet the problems persist. Any help would be appreciated.
1 (Deforum extension for auto1111 — version 2.2b | Git commit: ab8650a3 (Sat Mar 11 10:52:16 2023): UnboundLocalError: local variable 'value_is_number' referenced before assignment

2.RecursionError: maximum recursion depth exceeded while calling a Python object

The most recent crash just now crashed just 9 minutes into the job. That's the shortest one yet.
Not sure if this helps: File "C:\Users\dhaym\AppData\Roaming\Visions of Chaos\Examples\MachineLearning\Text To Image\stable-diffusion-webui\venv\lib\site-packages\torch\cuda_init_.py", line 160, in is_initialized
return _initialized and not _is_in_bad_fork()
RecursionError: maximum recursion depth exceeded while calling a Python object

RecursionError: maximum recursion depth exceeded while calling a Python object
Time taken: 9m 22.17sTorch active/reserved: 8069/9690 MiB, Sys VRAM: 12323/49140 MiB (25.08%)

Interesting thing is, this is intermittent - if I actually finish a job, I can most likley complete another one if I leave the UI open. But then after some time, it will crash again with one of the errors, using the "maximum recursion" one.

**Comment by MarkDedden** [comment]
The first error could be solved by opening helpers/animation.py, go down to the following code:
![image](https://user-images.githubusercontent.com/60524434/233859567-cd2a05f7-5e8c-41e5-97d1-f1ea9b335305.png)

Change the ident of the "if not value_in_number:" to be on the same spacing as the "if value_is_number".
![image](https://user-images.githubusercontent.com/60524434/233859672-62efc8cb-5dad-4fa4-9218-b81df487cc0b.png)

Save the file and restart your notebook. Now it should work.

About the second error you describe, recursiveness is usually when a function calls itself. The answer in this post has some good information on it and a solution that might work in your case. https://stackoverflow.com/questions/14222416/recursion-in-python-runtimeerror-maximum-recursion-depth-exceeded-while-callin

Try changing the recursion limit to 2000 and see if it runs longer.
 `sys.setrecursionlimit(2000)`

**Reply by dhmiller** [reply]
Thanks much. Not sure where I would access the code to change the recursion limit. Is that in one of the py files? I running under WEB UI...

**Reply by MarkDedden** [reply]
I assume in scripts/deforum.py directly below `import sys`.
After I made the previous post, I ran some recursions myself with a higher limit and I saw that at some point my Python crashed. So I am not sure if it will really solve the problem or if it just crashes.

**Reply by dhmiller** [reply]
K - well thanks for the advice - I'll try it and see what happens.
